### PR TITLE
Create Section 23 of the Controller Admin guide

### PR DESCRIPTION
**Pull request by EMcWhinn**
Added assembly and modules

Create section 23 of the Controller admin guide

https://issues.redhat.com/browse/AAP-17279

Affects `titles/controller-admin-guide`

### CODE REVIEW

File: downstream/assemblies/platform/assembly-ag-controller-LDAP-auth.adoc [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+Administrators use LDAP as a source for account authentication information for {ControllerName} users. 
+User authentication is provided, but not the synchronization of user permissions and credentials. 
+Organization membership, as well as the organization administrator, and team memberships can be synchronized.
+
+.LDAP authentication performance tips
+
+By default, when an LDAP user authenticates, all user-related attributes are updated in the database on each login. 
+In some environments, you can skip this operation due to performance issues. 
+To avoid it, you can disable the option *AUTH_LDAP_ALWAYS_UPDATE_USER*.
+For more information, see link:https://access.redhat.com/solutions/5823061[How can I prevent the LDAP attributes to be updated on each login?].
```
Comment (ianf77) [inline downstream/assemblies/platform/assembly-ag-controller-LDAP-auth.adoc:14]: That's dodgy wording for the title!

File: downstream/assemblies/platform/assembly-ag-controller-LDAP-auth.adoc [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+Note that new users are still created and get their attributes pushed to the database on their first login.
```
Comment (ianf77) [inline downstream/assemblies/platform/assembly-ag-controller-LDAP-auth.adoc:15]: users are created and their attributes are pushed

File: downstream/assemblies/platform/assembly-ag-controller-LDAP-auth.adoc [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+[WARNING]
+====
+With this option set to `False`, no changes to LDAP user's attributes are updated.
```
Comment (ianf77) [inline downstream/assemblies/platform/assembly-ag-controller-LDAP-auth.adoc:19]: Set this option to 'false' to not update the LDAP user's attributes ?

File: downstream/modules/platform/proc-controller-change-timeout-auth.adoc [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+. From the navigation panel, select menu:Settings[].
+. Select *Miscellaneous Authentication settings* from the list of *System* options.
+. Click btn:[Edit].
+. Enter the timeout period in seconds in the *Idle Time Force Log Out* text field.
+. Click btn:[Save].
+
+[NOTE]
+====
+If you access {ControllerName} and have trouble logging in, clear your web browser's cache. 
+In situations such as this, it is common that the authentication token has been cached in the browser session and must be cleared.
```
Comment (ianf77) [inline downstream/modules/platform/proc-controller-change-timeout-auth.adoc:18]: common for the authentication token to be cached during the browser session. It must cleared to continue.

File: downstream/modules/platform/proc-controller-set-up-LDAP.adoc [hunk 5]
```diff
@@ -0,0 +1,174 @@
+[id="controller-set-up-LDAP"]
+
+= Setting up LDAP authentication
+
+When configured, a user who logs in with an LDAP username and password automatically gets an {ControllerName} account created for them and they can be automatically placed into organizations as either regular users or organization administrators.
```
Comment (ianf77) [inline downstream/modules/platform/proc-controller-set-up-LDAP.adoc:5]: automatically has an

can can be automatically placed

File: downstream/modules/platform/proc-controller-set-up-LDAP.adoc [hunk 6]
```diff
@@ -0,0 +1,174 @@
...
+If the LDAP server you want to connect to has a certificate that is self-signed or signed by a corporate internal certificate authority (CA), 
+you must add the CA certificate to the system's trusted CAs. 
+Otherwise, connection to the LDAP server results in an error that the certificate issuer is not recognized. 
+For more information, see link:https://access.redhat.com/solutions/5136201[How Do I Import CA Certificate (openldap, AD, IDM, FreeIPA) Required for LDAPS Integration?]. 
+If prompted, use your Red Hat customer credentials to login.
+====
+
+.Procedure
+
+. Create a user in LDAP that has access to read the entire LDAP structure.
```
Comment (ianf77) [inline downstream/modules/platform/proc-controller-set-up-LDAP.adoc:21]: Does it say how to do this?

File: downstream/modules/platform/proc-controller-set-up-LDAP.adoc [hunk 7]
```diff
@@ -0,0 +1,174 @@
...
+. Use the `ldapsearch`` command to test if you can make successful queries to the LDAP server.
+This is a command line tool that you can install on the {ControllerName} system's command line as well as on other Linux and OSX systems.
```
Comment (ianf77) [inline downstream/modules/platform/proc-controller-set-up-LDAP.adoc:23]: You can install this tool fom {ControllerNames}'s system command line, and using other Linux and OSX systems.

File: downstream/modules/platform/proc-controller-set-up-LDAP.adoc [hunk 8]
```diff
@@ -0,0 +1,174 @@
...
++
+The equivalent API endpoints show `AUTH_LDAP_*` repeated: `AUTH_LDAP_1_*`, `AUTH_LDAP_2_*`, `AUTH_LDAP_5_*` to denote server designations.
+. To enter or modify the LDAP server address, click btn:[Edit] and enter in the *LDAP Server URI* field using the same format as the one pre-populated in the text field.
++
+[NOTE]
+====
+You can specify multiple LDAP servers by separating each with spaces or commas. Click the image:tooltips-icon.png[Tooltip] icon to comply with the correct syntax and rules.
+====
++
+. Enter the password to use for the binding user in the *LDAP Bind Password* text field.
```
Comment (ianf77) [inline downstream/modules/platform/proc-controller-set-up-LDAP.adoc:52]: These are set in the inventory file, just so you know :-)

File: downstream/modules/platform/proc-controller-set-up-LDAP.adoc [hunk 9]
```diff
@@ -0,0 +1,174 @@
...
++
+. Enter the password to use for the binding user in the *LDAP Bind Password* text field. 
+. Click to select a group type from the *LDAP Group Type* menu.
++
+The LDAP group types that are supported by {ControllerName} use the underlying link:https://django-auth-ldap.readthedocs.io/en/latest/groups.html#types-of-groups[django-auth-ldap library].
+To specify the parameters for the selected group type, see step 15.
+. The *LDAP Start TLS* is disabled by default. 
+To enable TLS when the LDAP connection is not using SSL, set the toggle to *On*.
+. Enter the distinguished name in the *LDAP Bind DN* text field to specify the user that {ControllerName} uses to connect (Bind) to the LDAP server. 
+*  If that name is stored in key `sAMAccountName`, the *LDAP User DN Template* populates with `(sAMAccountName=%(user)s)`.
```
Comment (ianf77) [inline downstream/modules/platform/proc-controller-set-up-LDAP.adoc:60]: Template* is populated from '(

File: downstream/modules/platform/proc-controller-set-up-LDAP.adoc [hunk 10]
```diff
@@ -0,0 +1,174 @@
...
+"last_name": "sn",
+"email": "mail"
+}
+----
++
+The previous example retrieves users by last name from the key `sn`. 
+You can use the same LDAP query for the user to determine what keys they are stored under.
++
+Depending on the selected *LDAP Group Type*, different parameters are available in the *LDAP Group Type Parameters* field to account for this. 
+`LDAP_GROUP_TYPE_PARAMS` is a dictionary that is converted by {ControllerName} to `kwargs` and passed to the *LDAP Group Type* class selected.
```
Comment (ianf77) [inline downstream/modules/platform/proc-controller-set-up-LDAP.adoc:142]: This one isn't an inventory  variable. It's in a file called ldap_extras.yml

File: downstream/modules/platform/proc-controller-set-up-LDAP.adoc [hunk 11]
```diff
@@ -0,0 +1,174 @@
...
+"is_system_auditor": "cn=auditors,ou=groups,dc=website,dc=com"
+}
+----
++
+. For more information on completing the mapping fields, *LDAP Organization Map* and *LDAP Team Map*, see the [ADD XREF]LDAP Organization and Team Mapping section.
+. Click btn:[Save].
+
+[NOTE]
+====
+{ControllerNameStart} does not actively sync users, but they are created during their initial login. To improve performance associated with LDAP authentication, see LDAP authentication performance tips at the end of this chapter.
```
Comment (ianf77) [inline downstream/modules/platform/proc-controller-set-up-LDAP.adoc:169]: synchronize

**Review by ianf77 — COMMENTED** [review]
Few comments

File: downstream/modules/platform/proc-controller-set-up-LDAP.adoc (hunk 6 above)
Comment (EMcWhinn) [reply downstream/modules/platform/proc-controller-set-up-LDAP.adoc:21]: I'll need to ask for SME input for the correct link or steps.

**Review by ianf77 — APPROVED** [review]
LGTM

**Pull request merged**